File: proverb/proverb.go
package proverb

import (
	"fmt"
	"strings"
)

func Proverb(words []string) string {
	phrases := []string{}
	for i := 0; i < len(words)-1; i++ {
		phrases = append(phrases, fmt.Sprintf("For want of a %s the %s was lost.", words[i], words[i+1]))
	}
	phrases = append(phrases, "And all for the want of a nail.")
	return strings.Join(phrases, "\n")
}

File: src/proverb.rs
pub fn proverb(words:Vec<&str>) -> String {
    let mut phrases = Vec::new();
    for i in 0..words.len()-1 {
        phrases.push(format!("For want of a {} the {} was lost.", words[i], words[i+1]));
    }
    phrases.push("And all for the want of a nail.".to_string());
    return phrases.join("\n");   
}